subsys/caf/modules/power_manager.c

```diff
@@ -222,6 +267,33 @@ static bool event_handler(const struct event_header *eh)
 		EVENT_SUBMIT(event);
 		return false;
 	}
+#if IS_ENABLED(CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET)
```

Comment:
preprocessor should be avoided in this case

if possible do:
```
   if (IS_ENABLED(CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET) &&
       is_power_manager_configure_timeout_event(eh)) {
   }
```

The above will make compiler always check this code (as preprocessor won't remove it). Compiler's optimizer will opt it out so the code check is for free :)


In cases where you need to remove the code in preprocessor.
In general I am not using IS_ENABLED in preprocessor and I think this is a common convention (though there are few cases where people use it). I don't think it's needed or gives any advantage.

If you used preprocessor because some option or variable was missing.
If this is a variable - you don't need to. Optimizer will remove the variable along with unused block of code.
If this is a config option. There are two ways of handling that - either define an option in source file if it was not defined in kconfig (I don't like it as there are two sources of option and it pollutes the source file) or add a option in Kconfig without prompt (that will make the option present but only to make preprocessor happy). We did that in several places (for an example see https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/src/modules/Kconfig.ble#L88 and https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/src/modules/Kconfig.ble#L44).